src/app/components/statistics/charts/chart-data-formatter/formatters/priority-chart-data.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable } from '@angular/core';
+import { ChartData } from 'chart.js';
+import { AllocationData } from 'src/app/shared/models/allocation-data';
+import { ChartDataFormatter } from '../chart-data-formatter';
+import { Student } from 'src/app/api/models';
+import { PeopleChartProjectData, PeopleChartStudentData } from '../people-chart-data';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PriorityChartDataService implements ChartDataFormatter {
+  private readonly PRIORITY_ZONE_LABELS = ['High Priority', 'Medium Priority', 'Low Priority', 'Unassigned'];
+  private readonly BACKGROUND_COLORS = ['#94DA7C', '#EED373', '#E16868', '#E0E0E0'];
+
+  getDoughnutData(allocationData: AllocationData): ChartData<'doughnut', number[], unknown> {
+    const priorityZoneCounts = new Map(this.PRIORITY_ZONE_LABELS.map(priority => [priority, 0]));
+
+    allocationData.projectsData.forEach(projectData => {
+      projectData.students.forEach(student => {
+        const priority = this.getStudentPriorityForProject(student, projectData.project.id);
+        const priorityZoneLabel = this.getPriorityZoneLabel(priority);
+        const currentCount = priorityZoneCounts.get(priorityZoneLabel);
+        priorityZoneCounts.set(priorityZoneLabel, currentCount + 1);
+      });
+    });
+
+    priorityZoneCounts.set(this.PRIORITY_ZONE_LABELS[3], allocationData.studentsWithoutTeam.length);
+
+    return {
+      labels: this.PRIORITY_ZONE_LABELS,
+      datasets: [
+        {
+          data: Array.from(priorityZoneCounts.values()),
+          backgroundColor: this.BACKGROUND_COLORS,
+        },
+      ],
+    };
+  }
+
+  getPeopleData(allocationData: AllocationData): PeopleChartProjectData[] {
+    const data: PeopleChartProjectData[] = [];
+
+    allocationData.projectsData.forEach(projectData => {
+      const studentData: PeopleChartStudentData[] = [];
+      const studentPriorities: number[] = [];
+
+      projectData.students
```

Comment:
I would split this into multiple functions. `.map()`,  `.sort()` +  `.forEach() ` is a lot. I would assign the return of `.map()` to s extra const.